src/w2grid.js

```diff
@@ -701,7 +702,8 @@
             // process date fields
             obj.selectionSave();
             obj.prepareData();
-            obj.reset();
+            if (!noReset)
+                obj.reset();
```

Comment:
I would try to avoid adding more parameters to sort functions, if possible. What exactly in reset method that messes up sort?

Also if "if" statement does not fit one lin, please use {}